src-tauri/src/menu.rs

```diff
@@ -0,0 +1,48 @@
+use tauri::{CustomMenuItem, Menu, SystemTray, SystemTrayMenu, SystemTrayMenuItem};
+
+#[cfg(target_os = "macos")]
+use tauri::{MenuItem, Submenu};
+
+pub fn get_tray_menu() -> SystemTray {
+    let quit = CustomMenuItem::new("quit".to_string(), "Quit");
+    let toggle_visibility = CustomMenuItem::new("toggle_visibility".to_string(), "Hide");
+    let tray_menu = SystemTrayMenu::new()
+        .add_item(toggle_visibility)
+        .add_native_item(SystemTrayMenuItem::Separator)
+        .add_item(quit);
+
+    SystemTray::new().with_menu(tray_menu)
+}
+
+pub fn get_menu() -> Menu {
+    #[cfg(target_os = "macos")]
+    {
+        let my_app_menu = Menu::new()
+            .add_native_item(MenuItem::Hide)
+            .add_native_item(MenuItem::Separator)
+            .add_native_item(MenuItem::Quit);
+
+        let edit_menu = Menu::new()
+            .add_native_item(MenuItem::Undo)
+            .add_native_item(MenuItem::Redo)
+            .add_native_item(MenuItem::Separator)
+            .add_native_item(MenuItem::Cut)
+            .add_native_item(MenuItem::Copy)
+            .add_native_item(MenuItem::Paste)
+            .add_native_item(MenuItem::SelectAll);
+
+        let window_menu = Menu::new()
+            .add_native_item(MenuItem::Minimize)
+            .add_native_item(MenuItem::CloseWindow);
```

Comment:
Why do we need `edit` menu? Also it seems that `my_app` and `window` are very similar if not the same.

Reply:
We need edit menu, since if we do not add these native items, we cannot use the shortcuts on mac. Unfortunately that's how it works. And if they all populate the same submenu, it looks really really bad. 
So the convention for macos applications is storing these (cut, copy, etc.) under edit menu
<img width="614" alt="image" src="https://user-images.githubusercontent.com/32795992/161395440-a9f19780-00fe-413c-9695-b3b8afe68e41.png">
this is also the case in the other apps I've used like discord etc. I just followed the convention for being more user-friendly

Reply:
Hm, edit menu makes sense, though this is definitely a strange macOS design decision. What about App vs Window?

Reply:
It is actually the same reason, following the convention. I agree with you they are being similar `App` and `Window`. But for **minimize, zoom, etc.** these are stored in `Window` submenu. On the other hand, **hide and quit** are stored in the main submenu, which is the app itself (**subspace-desktop** in our case). 
<img width="1242" alt="image" src="https://user-images.githubusercontent.com/32795992/161418286-7e0b2ea3-489d-4c63-a788-508257a7f848.png">
<img width="322" alt="image" src="https://user-images.githubusercontent.com/32795992/161418298-e5127ed1-0354-4734-9395-e224eae1fcc0.png">